refactor(VLEComponent): Remove UpgradeModule dependency

hirokiterashima opened the pull request:
## Changes
- Remove UpgradeModule from VLEComponent
- Remove hybrid AngularJS-Angular support from VLE and Preview. They are now fully Angular
- Remove unused dependencies
- Rename classes to reflect new changes

## Test Prep
- Run npm install to remove dependencies we no longer need

## Test
- Student VLE and Preview work as before

hirokiterashima committed:
refactor(StudentDataService): Remove UpgradeModule dependency

hirokiterashima committed:
chore(i18n): Generate messages

hirokiterashima committed:
refactor(VLEComponent): Remove UpgradeModule dependency

hirokiterashima committed:
Merge branch 'develop' into issue-651-remove-upgrade-module-from-vle-component

hirokiterashima committed:
chore(StudentVLE): Convert to full Angular

- Remove UpgradeModule from VLEComponent
- Remove hybrid support from StudentVLE

#651

hirokiterashima committed:
chore(StudentVLE): Convert to Angular

- Fix change detection errors thrown in development mode

#651

geoffreykwan commented on src/assets/wise5/services/initializeVLEService.ts:
```diff
@@ -0,0 +1,68 @@
+    private studentWebSocketService: StudentWebSocketService
+  ) {}
+
+  initializeStudent(unitId: string) {
```
Perhaps replace the promise chains with async/await.

```
  async initializeStudent(unitId: string) {
    await this.configService.retrieveConfig(`/api/config/studentRun/${unitId}`);
    this.sessionService.initializeSession();
    this.studentStatusService.retrieveStudentStatus();
    await this.projectService.retrieveProject();
    this.studentWebSocketService.initialize();
    await this.studentDataService.retrieveStudentData();
    await this.notificationService.retrieveNotifications();
    await this.achievementService.retrieveStudentAchievements();
    await this.studentDataService.retrieveRunStatus();
    await this.studentAssetService.retrieveAssets();
    await this.notebookService.retrieveNotebookItems(this.configService.getWorkgroupId());
    this.intializedSource.next(true);
  }

  async initializePreview(unitId: string) {
    await this.configService.retrieveConfig(`/api/config/preview/${unitId}`);
    this.sessionService.initializeSession();
    this.studentStatusService.retrieveStudentStatus();
    await this.projectService.retrieveProject();
    this.studentDataService.retrieveStudentData();
    this.studentDataService.retrieveRunStatus();
    this.notificationService.retrieveNotifications();
    this.intializedSource.next(true);
  }
```

geoffreykwan commented on src/assets/wise5/components/openResponse/open-response-student/open-response-student.component.html:
```diff
@@ -85,10 +85,9 @@
+    <div *ngIf="attachment.type === 'audio'" fxLayout="row" fxLayoutAlign="start center">
       <audio controls>
-        <source [src]="attachment.url | safeUrl" type="audio/webm">
+        <source src="{{attachment.url}}" type="audio/webm">
```
Is there a reason why we can't use [src] here? I tried it and it seemed to work.

geoffreykwan reviewed (changes requested):
Everything seems to work properly like before. I noted two minor things we could change.

hirokiterashima committed:
chore(StudentVLE): Convert to Angular

- Replace promises chaining with aync/await
- Add [src] binding to OR audio source element

#651

geoffreykwan committed:
fix(Student VLE): Make VLE use full height

Previously if the content did not take up the full height of the page
the VLE would not stretch to the bottom of the page. Now it does use
up the full height no matter what.

#651

geoffreykwan reviewed (approved):
Looks good.
